specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/mfe.json

```diff
@@ -0,0 +1,1719 @@
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/APIVersionParameter"
+          },
+          {
+            "in": "path",
+            "name": "name",
+            "description": "Datastore name.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Datastore entity to create or update.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DatastorePropertiesResource"
+            }
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "201": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DatastorePropertiesResource"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create or update datastore.": {
+            "$ref": "./examples/createOrUpdateDatastore.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/datastores/{name}/listSecrets": {
+      "post": {
+        "tags": [
+          "Datastore"
+        ],
+        "summary": "Get datastore secrets.",
+        "operationId": "Datastores_ListSecrets",
+        "parameters": [
+          {
+            "$ref": "machineLearningServices.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/APIVersionParameter"
+          },
+          {
+            "in": "path",
+            "name": "name",
+            "description": "Datastore name.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DatastoreCredentials"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get datastore secrets.": {
+            "$ref": "./examples/listDatastoreSecrets.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/datastores/{name}/setAzureMLDefault": {
+      "post": {
```

Comment:
Why is setting the default modeled as a POST action instead of a property on the datastore?  Modeling it as POST will prevent users from setting default in a template.

Reply:
This is because this new ARM API is a mapping over our existing non-ARM API, which manages the default through separate actions like this.

Reply:
Removed this POST operation; instead the isDefault property is writeable, so this will be handled via the normal PUT endpoint.